cgi/kratos_update_settings.pl

```diff
@@ -0,0 +1,131 @@
+    my $req = HTTP::Request->new(GET => $url);
+    $req->header('content-type' => 'application/json');
+    $req->header('Cookie' => $kratos_cookie);
+
+    my $resp = $ua->request($req);
+
+    if ($resp->is_success) {
+        #decode json to a hash
+        my $json = $resp->decoded_content;
+        my $content_ref = decode_json($json);
+
+        #get user info from kratos json hash
+        my $UserID = $content_ref->{identity}{traits}{UserID};
+        my $name_kratos = $content_ref->{identity}{traits}{name};
+        my $email_kratos = $content_ref->{identity}{traits}{email};
+        my $newsletter_kratos = $content_ref->{identity}{traits}{newsletter};
+        my $edit_link_kratos = $content_ref->{identity}{traits}{"Add Edit Link"};
+        my $display_barcode_kratos = $content_ref->{identity}{traits}{"Display Barcode"};
+        my $team_1_kratos = $content_ref->{identity}{traits}{Teams}{"Team 1"};
+        my $team_2_kratos = $content_ref->{identity}{traits}{Teams}{"Team 2"};
+        my $team_3_kratos = $content_ref->{identity}{traits}{Teams}{"Team 3"};
+        my $professional_account_kratos = $content_ref->{identity}{traits}{"professional_account"};
+
+        #retrieve users storable file
+        my $user_file = "$data_root/users/" . get_string_id_for_lang("no_language", $UserID) . ".sto";
+        my $user_ref = retrieve($user_file);
+
+        $user_ref->{userid} => $UserID;
+        $user_ref->{email} => $email_kratos;
+        $user_ref->{name} => $name_kratos;
+
+        #updating users info
+        if($newsletter_kratos == 0){
+            $user_ref->{newsletter} = '';
+        }
+        else{
+            $user_ref->{newsletter} = 'on';
+        }
+
+        if($edit_link_kratos == 0){
+            $user_ref->{edit_link} = '';
+        }
+        else{
+            $user_ref->{edit_link} = 1;
+        }
+
+        if($display_barcode_kratos == 0){
+            $user_ref->{display_barcode} = '';
+        }
+        else{
+            $user_ref->{display_barcode} = 1;
+        }
+
+        if($team_1_kratos ne ''){
+            $user_ref->{team_1} = $team_1_kratos;
+        }
+
+        if($team_2_kratos ne ''){
+            $user_ref->{team_2} = $team_2_kratos;
+        }
+
+        if($team_3_kratos ne ''){
+            $user_ref->{team_3} = $team_3_kratos;
+        }
+
+        #update only if user does not already have a pro account
+        if($professional_account_kratos ne '' and $user_ref->{pro} ne 1){
+            $user_ref->{org} = $professional_account_kratos;
+            $user_ref->{org_id} = $professional_account_kratos;
+            $user_ref->{pro} = 1;
+        }
+
+        #store the updated sto file
+        store($user_ref, $user_file);
+    }
+    else {
+        $log->debug("HTTP GET error code: ", $resp->code, "n");
+        $log->debug("HTTP GET error message: ", $resp->message, "n");
+    }
+}
+
+print redirect(-url=>'http://world.openfoodfacts.localhost/');
```

Comment:
I think you should use: 

```perl
redirect_to_url($request_ref, 302, 'http://world.openfoodfacts.localhost/');
```
But of course the url should normally be built (but I imagine it was for testing).